kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/FilterIT.java

```diff
@@ -293,74 +291,6 @@ void supportsForwardDeferredByAsynchronousBrokerRequest(RequestResponseMarkingFi
                 ForwardingStyle.ASYNCHRONOUS_REQUEST_TO_BROKER);
     }
 
-    /**
-     * This is here to test a subtle case, we want to ensure that if a Filter defers some work
-     * and completes that work from some uncontrolled thread, that other Messages enqueued at the
-     * filter are dispatched on the eventloop. Given that the eventloop is dynamically allocated
-     * to the channel, we cannot precisely check what thread we are running on, but we can
-     * assert that the user's Filter method is being invoked from the same thread.
-     */
-    @ParameterizedTest
-    @EnumSource(value = RequestResponseMarkingFilterFactory.Direction.class)
-    void supportAsyncBatchForwarding(RequestResponseMarkingFilterFactory.Direction direction) throws Exception {
```

Comment:
for context, this whole test is replaced by an invariant assertion in RequestResponseMarkingFilter, which returns a failed future if any of it's methods are invoked by a thread other than the dispatch thread